psamm_import/importer.py

```diff
@@ -116,19 +116,68 @@ def model_compounds(model):
         yield d
 
 
-def model_reactions(model, exchange=False):
-    """Yield model reactions as YAML dicts"""
+def reactions_to_files(model, dest, yaml_args, exchange):
+    """Turns the reaction subsystems into their own files.
 
+    If a subsystem has a number of reactions over the threshold, it gets its
+    own yaml file. All other reactions, those that don't have a subsystem or
+    are in a subsystem that falls below the threshold, get added to a common
+    reaction file.
+    """
+
+    common = []
+    all_subsys_ids = []
+    all_subsys = []
+    subsys_files = []
     for reaction_id, reaction in sorted(iteritems(model.reactions)):
+        subsys = reaction.subsystem
+        r = reaction
+        if(subsys is not None and subsys not in all_subsys_ids and
+                len(r.equation.compounds) != 1):
+            all_subsys_ids.append(subsys)
+            all_subsys.append([r])
+        elif subsys in all_subsys_ids:
+            subsys_index = all_subsys_ids.index(subsys)
+            all_subsys[subsys_index].append(r)
+        elif len(r.equation.compounds) != 1:
+            common.append(r)
+
+    for subsys in all_subsys:
+            if len(subsys) < 3:
+                for r in subsys:
+                    common.append(r)
+                all_subsys.remove(subsys)
+
+    common = list(model_reactions(common, exchange=exchange))
+    if len(all_subsys) > 0:
+        for subsys in all_subsys:
+            subsysfile = str(subsys[1].subsystem).replace('/', '_') + '.yaml'
```

Comment:
We need to make sure that this is a valid file name on the current platform. Does Python provide a method to ensure that a file name is valid? Maybe we should also make some additional conversions such as converting to lower case and removing space characters?